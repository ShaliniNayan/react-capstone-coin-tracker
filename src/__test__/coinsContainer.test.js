import { MemoryRouter } from 'react-router-dom';
import { Provider, useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../redux/Store';
import CoinsContainer from '../components/CoinsContainer';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

const data = [
  {
    name: 'Bitcoin',
    symbol: 'BTC',
    rank: 1,
    id: 'bitcoin',
    price: '30048.40',
    percent_change_24h: '-0.31',
    percent_change_1h: '0.11',
    percent_change_7d: '-1.59',
    price_btc: '1.00',
    market_cap_usd: '583163759527.79',
    volume24: 18382660443.609882,
    volume24a: 10487103713.130768,
    supply: '19407484.00',
  },
  {
    name: 'Ethereum',
    symbol: 'ETH',
    rank: 2,
    id: '80',
    price: '1857.17',
    percent_change_24h: '0.21',
    percent_change_1h: '0.24',
    percent_change_7d: '-1.28',
    price_btc: '0.061798',
    market_cap_usd: '227272062947.10',
    volume24: 18382660443.609882,
    volume24a: 10793675580.448414,
    supply: '122375302.00',
  },
];

describe('CoinContainer', () => {
  beforeEach(() => {
    useSelector.mockReturnValue(data);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CoinsContainer />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
