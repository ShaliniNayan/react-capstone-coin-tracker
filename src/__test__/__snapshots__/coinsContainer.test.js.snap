// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinContainer Should renders correctly 1`] = `
<div>
  <div
    class="search"
  >
    <input
      placeholder="Search Coins"
      type="search"
      value=""
    />
  </div>
  <div
    class="coins-wrap"
  >
    <h2
      class="title"
    >
      Available Coins
    </h2>
    <div
      class="collections"
    >
      <a
        class="single-collection"
        href="/details"
      >
        <div>
          <div>
            <h2
              class="coin-name"
            >
              Crypto Name =
               
              Bitcoin
            </h2>
          </div>
          <div
            class="coin-info-wrap"
          >
            <span>
              Symbol =
               
              BTC
            </span>
            <span>
              Price =
               
              30048.40
            </span>
            <span
              class="ranking"
            >
               
              Crypto Ranking =
              #1
              <i>
                <img
                  alt=""
                  class="stats-icon"
                  src="stats1.png"
                />
              </i>
            </span>
          </div>
        </div>
      </a>
      <a
        class="single-collection"
        href="/details"
      >
        <div>
          <div>
            <h2
              class="coin-name"
            >
              Crypto Name =
               
              Ethereum
            </h2>
          </div>
          <div
            class="coin-info-wrap"
          >
            <span>
              Symbol =
               
              ETH
            </span>
            <span>
              Price =
               
              1857.17
            </span>
            <span
              class="ranking"
            >
               
              Crypto Ranking =
              #2
              <i>
                <img
                  alt=""
                  class="stats-icon"
                  src="stats1.png"
                />
              </i>
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
